packages/ai/src/ui/abstract-chat.ts

```diff
@@ -83,14 +54,11 @@ export type InferUIDataParts<T extends UIDataPartSchemas> = {
 };
 
 export interface ChatState<MESSAGE_METADATA, DATA_TYPES extends UIDataTypes> {
-  getStatus: () => ChatStatus;
-  setStatus: (status: ChatStatus) => void;
+  status: ChatStatus;
 
-  getError: () => Error | undefined;
-  setError: (error: Error | undefined) => void;
+  error: Error | undefined;
 
-  getMessages: () => UIMessage<MESSAGE_METADATA, DATA_TYPES>[];
-  setMessages: (messages: UIMessage<MESSAGE_METADATA, DATA_TYPES>[]) => void;
```

Comment:
This saves us a bunch of code in the library implementations -- if a library needs to run "middleware" on gets and sets they can just declare this as a getter and setter pair, which retains the same functionality